### PR TITLE
fix error (doc:fetch-data-via-sdk.mdx)

### PR DESCRIPTION
**Pull request by caoyang2002**
### Description

The second code block example, which is labeled as `fetch-data.ts`; correct the erroneous syntax to a nested object.

### Checklist

The `apps/nextra/pages/en/build/sdks/ts-sdk/fetch-data-via-sdk.mdx`

> appendix:

```ts
queryIndexer<T extends {}>(args: {
    query: GraphqlQuery;
}): Promise<T>;
```

```
type GraphqlQuery = {
    query: string;
    variables?: {};
};
```

- Do all Lints pass?
  - [ ] Have you ran `pnpm spellcheck`?
  - [ ] Have you ran `pnpm fmt`?
  - [ ] Have you ran `pnpm lint`?

### CODE REVIEW

**Comment by WGB5445** [comment]
Very good. Thank you very much.

File: apps/nextra/pages/en/build/sdks/ts-sdk/fetch-data-via-sdk.mdx [hunk 1]
```diff
@@ -22,13 +22,16 @@ Many queries have a parameter named `options` to customize the results, use it t
...
 
 ```ts filename="fetch-data.ts"
-const ledgerInfo = aptos.queryIndexer({
- query: `query MyQuery {
-     ledger_infos {
-       chain_id
-     }
-   }`;
-})
+  const ntf_collection_info = aptos.queryIndexer({
```
Comment (banool) [inline apps/nextra/pages/en/build/sdks/ts-sdk/fetch-data-via-sdk.mdx:25]: This should still be called `ledgerInfo` yeah?

**Review by banool — CHANGES_REQUESTED** [review]

File: apps/nextra/pages/en/build/sdks/ts-sdk/fetch-data-via-sdk.mdx (hunk 1 above)
Comment (0xmaayan) [inline apps/nextra/pages/en/build/sdks/ts-sdk/fetch-data-via-sdk.mdx:25]:
```suggestion
  const ledgerInfo = aptos.queryIndexer({
```

File: apps/nextra/pages/en/build/sdks/ts-sdk/fetch-data-via-sdk.mdx [hunk 2]
```diff
@@ -22,13 +22,16 @@ Many queries have a parameter named `options` to customize the results, use it t
...
+    query: {
+      query: `query GetLedgerInfo {
+        ledger_infos {
+          chain_id
+        }
+      }`,
+      variables: {}, // If the query does not require variables, it can be an empty object here.
```
Comment (0xmaayan) [inline apps/nextra/pages/en/build/sdks/ts-sdk/fetch-data-via-sdk.mdx:32]: If the query does not require variables, we can simply not add a `variables` field

https://github.com/aptos-labs/aptos-ts-sdk/blob/main/tests/e2e/api/general.test.ts#L224

**Comment by 0xmaayan** [comment]
> ### Description
> The second code block example, which is labeled as `fetch-data.ts`; correct the erroneous syntax to a nested object.
> 
> ### Checklist
> The `apps/nextra/pages/en/build/sdks/ts-sdk/fetch-data-via-sdk.mdx`
> 
> > appendix:
> 
> ```ts
> queryIndexer<T extends {}>(args: {
>     query: GraphqlQuery;
> }): Promise<T>;
> ```
> 
> ```
> type GraphqlQuery = {
>     query: string;
>     variables?: {};
> };
> ```
> 
> * Do all Lints pass?
>   
>   * [ ]  Have you ran `pnpm spellcheck`?
>   * [ ]  Have you ran `pnpm fmt`?
>   * [ ]  Have you ran `pnpm lint`?

Thanks for the PR! The issue is with the object syntax with a missing `query` field. The `variables` field is not required if the query does not require variables

**Review by 0xmaayan — APPROVED** [review]

**Review by gregnazario — APPROVED** [review]

**Comment by gregnazario** [comment]
Looks like all of these changes have gotten in already.  Thanks for the contribution!

**Pull request closed without merging**